lib/tasks/update_hb_services.rake

```diff
@@ -39,52 +39,96 @@ task :add_eap_id => :environment do
   def header
     [
-     'service_id',
-     'name',
-     'cpt_code',
-     'eap_id',
-     'revenue_code',
-     'is_available',
-     'organization_id',
-     'line_items_count',
-     'description'
+     'Service ID',
+     'EAP ID',
+     'CPT Code',
+     'Revenue Code',
+     'Procedure Name',
+     'Service Rate',
+     'Corporate Rate',
+     'Federal Rate',
+     'Member Rate',
+     'Other Rate',
+     'Is One Time Fee?',
+     'Clinical Qty Type',
+     'Unit Factor',
+     'Qty Min',
+     'Display Date',
+     'Effective Date'
     ]
   end
 
+  def update_service_pricing(service, row)
+    pricing_map = service.pricing_maps.build( full_rate: Service.dollars_to_cents(row['Service Rate'].to_s),
+                                              corporate_rate: Service.dollars_to_cents(row['Corporate Rate'].to_s),
+                                              federal_rate: Service.dollars_to_cents(row['Federal Rate'].to_s),
+                                              member_rate: Service.dollars_to_cents(row['Member Rate'].to_s),
+                                              other_rate: Service.dollars_to_cents(row['Other Rate'].to_s),
+                                              unit_factor: row['Unit Factor'],
+                                              unit_type: row['Clinical Qty Type'],
+                                              unit_minimum: row['Qty Min'],
+                                              units_per_qty_max: service.current_effective_pricing_map.units_per_qty_max,
+                                              quantity_type: service.current_effective_pricing_map.quantity_type,
+                                              otf_unit_type: service.current_effective_pricing_map.otf_unit_type,
+                                              quantity_minimum: service.current_effective_pricing_map.quantity_minimum,
+                                              display_date: Date.strptime(row['Display Date'], "%m/%d/%y"),
+                                              effective_date: Date.strptime(row['Effective Date'], "%m/%d/%y")
+                                              )
+    if pricing_map.valid?
+      pricing_map.save
+    else
+      puts "#"*50
+      puts "Error importing pricing map"
+      puts service.inspect
+      puts pricing_map.inspect
+      puts service.errors.inspect
+      puts pricing_map.errors.inspect
+    end
+  end
+
   revenue_codes = []
   cpt_codes = []
   service_names = []
+  pricing_maps = []
   puts ""
   puts "Reading in file..."
   input_file = Rails.root.join("db", "imports", get_file)
   continue = prompt('Preparing to modify the services. Are you sure you want to continue? (y/n): ')
 
-  if continue == 'y'
+  if (continue == 'y') || (continue == 'Y')
     ActiveRecord::Base.transaction do
       CSV.foreach(input_file, :headers => true) do |row|
-        service = Service.find(row['service_id'].to_i)
+        service = Service.where(id: row['Service ID'].to_i).first
```

Comment:
The problem is with the import file. There a bunch of null values after the last service row. The 'where' handles this.